### DISCUSSION TITLE
Demo broken again

### DISCUSSION THREAD

**Discussion by TronCarter269** [rzeldent/esp32-smartdisplay#209] [General]
This morning I tried to start from zero again and found that PlatformIO is installing lvgl 9.2.1, which once again breaks the demo.  Is there a way to force a library version?  Can the library.json file be downloaded and modified to run locally or something like that?

`Compiling .pio\build\esp32-3248S035C\lib127\esp32_smartdisplay\lvgl_panel_st7789_i80.c.o
.pio/libdeps/esp32-3248S035C/esp32_smartdisplay/src/esp32_smartdisplay.c: In function 'smartdisplay_init':
.pio/libdeps/esp32-3248S035C/esp32_smartdisplay/src/esp32_smartdisplay.c:197:15: error: 'lv_display_t' {aka 'struct lv_display_t'} has no member named 'sw_rotate'
   if (!display->sw_rotate)
               ^~
Compiling .pio\build\esp32-3248S035C\lib127\esp32_smartdisplay\lvgl_panel_st7789_spi.c.o
Compiling .pio\build\esp32-3248S035C\lib127\esp32_smartdisplay\lvgl_panel_st7796_spi.c.o
*** [.pio\build\esp32-3248S035C\lib127\esp32_smartdisplay\esp32_smartdisplay.c.o] Error 1
.pio/libdeps/esp32-3248S035C/esp32_smartdisplay/src/lvgl_panel_st7796_spi.c: In function 'lvgl_lcd_init':
.pio/libdeps/esp32-3248S035C/esp32_smartdisplay/src/lvgl_panel_st7796_spi.c:41:12: error: 'lv_display_t' {aka 'struct lv_display_t'} has no member named 'sw_rotate'
     display->sw_rotate = 0;
            ^~
*** [.pio\build\esp32-3248S035C\lib127\esp32_smartdisplay\lvgl_panel_st7796_spi.c.o] Error 1
`

If version control can't be accomplished, it is a pointless waste of (more) time to try to develop anything with this code.

**Comment by rzeldent** [comment]
Yes, you can fix the version by adding ^9.2.0 to the end of the library. Which branch are you using?
Seems like LVGL is changing the rotation, but documentation from them is lacking...

**Comment by TronCarter269** [comment]
Like this?
`lib_deps =
    https://github.com/rzeldent/esp32-smartdisplay#develop^9.2.0
`
I'm using the main branch of the demo.

**Reply by rzeldent** [reply]
No, it has to be changed in the dependencies of the library...

**Reply by silvioviscuso** [reply]
Hi, today I received the esp display, unfortunately I have the same error, I'm from a mac, I tried all the versions of lvgl and espressif and nothing to do...

**Comment by rzeldent** [comment]
Should be fixed now: See commit 596e2eaa03cbae3dacf6277a33397bf92483ce7a

**Reply by TronCarter269** [reply]
This worked for the demo platformio.ini:
 ```
lib_deps =
    https://github.com/rzeldent/esp32-smartdisplay#develop
    lvgl/lvgl@^9.2.0
```

**Reply by lazyoldjack** [reply]
Thanks for fixing this. I was pulling my hair out trying to get to the bottom of the problem. Moving forwards again now :o)
